concordium-node/build.rs

```diff
@@ -158,6 +158,11 @@ fn link_ghc_libs() -> std::io::Result<std::path::PathBuf> {
         ]))
     });
     let rts_dir = Path::new(&ghc_lib_dir).join("rts");
+    let rts_dir = if rts_dir.exists() {
+        rts_dir
+    } else {
+        Path::new(&ghc_lib_dir).join(GHC_VARIANT)
+    };
```

Comment:
Even if this works on all platforms it needs documentation as to why we are doing this.

I am also not a fan of this solution. It works by accident. It would be better to be explicit
- if linux do the first thing
- if intel mac do ...
- if M1 mac do the third thing

The way it is now is very indirect so hard to understand.